src/go/k8s/pkg/resources/configmap.go

```diff
@@ -121,18 +126,73 @@ func (r *ConfigMapResource) Ensure(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("error while fetching ConfigMap resource: %w", err)
 	}
-	_, err = Update(ctx, &cm, obj, r.Client, r.logger)
+
+	return r.update(ctx, &cm, obj.(*corev1.ConfigMap), r.Client, r.logger)
+}
+
+func (r *ConfigMapResource) update(
+	ctx context.Context,
+	current *corev1.ConfigMap,
+	modified *corev1.ConfigMap,
+	c k8sclient.Client,
+	logger logr.Logger,
+) error {
+	// Do not touch existing last-applied-configuration (it's not reconciled in the main loop)
+	if val, ok := current.Annotations[LastAppliedConfigurationAnnotationKey]; ok {
+		if modified.Annotations == nil {
+			modified.Annotations = make(map[string]string)
+		}
+		modified.Annotations[LastAppliedConfigurationAnnotationKey] = val
+	}
```

Comment:
No, probably you missed that the target is different: `current` vs `modified`. The meaning is that if the configmap in the cluster contains a value for that annotation, then the configmap we're going to apply will have the same value (since we use .Update instead of .Patch, we need to provide a full configmap when we apply it).